logs/driver.go

```diff
@@ -0,0 +1,41 @@
+package logs
+
+import "time"
+
+type Driver interface {
+	Send(l *Entry)
+	GetHelper() func()
```

Comment:
The GetHelper function has no documentation and it's not clear what it should for a user.
Because it's the interface for the Log Driver it should be clear.


I also suggest to change it either too:
```suggestion
	CallTestHelper()
```
Or 

```suggestion
        // PreLogFn returns a function that is called before each logging call is made.
        // It can be e.g. used in tests for blablubb...
	PreLogFn() func()
```

Reply:
I documented the function and renamed as suggested.

The first suggestion won't work. I imagine you didn't have enough information to understand the real reason for this function. The test logger (`logtotest` package) must have a way of using the `t.Helper` function in a way that the test log entries point to the code calling the logger, not to the logger. This is the reason for a second-order function. The driver can then `return t.Helper`. Example of outputs:

```go
func TestLog(t *testing.T) {
	logger := logs.New(logtotest.ForTest(t, false), // this is line 27 of logger_test.go
		logs.WithPrefix("TestLogger"),
		logs.WithDebugEnabledFn(func() bool { return true }))

	logger.I("Info", logs.WithString("key", "value"))
}

// Output of go test -v:
//     logger_test.go:27: [warn] TestLogger: Warn
//        - key: value
```

Observe that even though the `logs` test package called `t.Log()` the log message points to the line that invoked the logger, due to the usage of the helper. I know is a small use case, but I consider it important.

Reply:
Thanks for the explanation.

The usecase for t.Helper() was clear but it did not come to my mind that it won't work anymore with my suggestion. :-)